layerindex/tools/fedora-fetch.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+
+# Fedora Pagure (e.g. https://src.fedoraproject.org/) fetch utility
+
+# Copyright (C) 2017, 2018 Intel Corporation
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+import sys
+import os
+import argparse
+import json
+import requests
+import re
+import subprocess
+
+
+def get_repos_api(url, toplevel_item=None, name_attr='name', use_link_header=True, link_item='pagination', link_next_attr='next'):
+    link_re = re.compile('<(http[^>]+)>; rel="([a-zA-Z0-9]+)"')
+    repos = None
+    while True:
+        session = requests.Session()
+        print('getting %s' % url)
+        with requests.Session() as s:
+            r = s.get(url)
+            if not r.ok:
+                print('Request failed: %d: %s' % (r.status_code, r.text))
+                sys.exit(2)
+            st = r.text
+            jdata = json.loads(st)
+            if toplevel_item:
+                repos = jdata[toplevel_item]
+            else:
+                repos = jdata
+            for repo in repos:
+                name = repo[name_attr]
+                yield repo
+
+            link_url = None
+            if use_link_header:
+                link = r.headers.get('Link', None)
+                if link:
+                    linkitems = dict([reversed(x) for x in link_re.findall(link)])
+                    link_url = linkitems.get('next', None)
+            else:
+                link = jdata.get(link_item, None)
+                if link:
+                    link_url = link.get(link_next_attr, None)
+            if not link_url:
+                break
+            url = link_url
+
+
+def fetchall(args):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 23. By comparison, 99% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 127-137 into a separate function.